rom/dev/src/flow/cold_reset/ocp_lock.rs

```diff
@@ -4,8 +4,77 @@ use caliptra_api::mailbox::{
     ReportHekMetadataReq, ReportHekMetadataResp, ReportHekMetadataRespFlags,
 };
 use caliptra_cfi_lib::{cfi_assert, cfi_assert_bool, cfi_launder};
-use caliptra_common::keyids::ocp_lock::KEY_ID_HEK;
-use caliptra_drivers::{CaliptraResult, HekSeedState, KeyVault, Lifecycle, PersistentData, SocIfc};
+use caliptra_common::{
+    crypto::Crypto,
+    keyids::{
+        ocp_lock::{KEY_ID_HEK, KEY_ID_MDK},
+        KEY_ID_ROM_FMC_CDI,
+    },
+};
+use caliptra_drivers::{
+    CaliptraResult, HekSeedState, HmacMode, KeyUsage, KeyVault, Lifecycle, PersistentData, SocIfc,
+};
+
+use crate::rom_env::RomEnv;
+
+use zerocopy::IntoBytes;
+
+/// Run the OCP LOCK Cold Reset flow
+///
+/// # Arguments
+///
+/// * `env` - ROM Environment
+pub fn ocp_lock_cold_reset_flow(env: &mut RomEnv) -> CaliptraResult<()> {
+    if cfi_launder(!env.soc_ifc.ocp_lock_enabled()) {
+        return Ok(());
+    } else {
+        cfi_assert!(env.soc_ifc.ocp_lock_enabled())
+    }
+    derive_hek(env)?;
+    derive_mdk(env)?;
+    Ok(())
+}
+
+/// Derive OCP LOCK HEK
+///
+/// NOTE: The HEK _may_ be disabled once MCU has reported the HEK seed state.
+///
+/// # Arguments
+///
+/// * `env` - ROM Environment
+fn derive_hek(env: &mut RomEnv) -> CaliptraResult<()> {
+    let hek_seed = env.soc_ifc.fuse_bank().ocp_hek_seed();
+    Crypto::hmac_kdf(
+        &mut env.hmac,
+        &mut env.trng,
+        KEY_ID_ROM_FMC_CDI,
```

Comment:
I'm not sure I realized this previously.

Do we want to use the FMC CDI here? That would mean an update to FMC firmware will crypto erase the drive. That is a pretty scary side effect people are sure to forget about.